### PR TITLE
(DOCSP-2437): Added Kerberos options.

### PR DESCRIPTION
**Pull request by atsansone**
@rychipman : I have provided all of the options. I still have a question posted to you in JIRA for DOCSP-2437 about where in the conf file these values would go. Please have a look. 

- [Kerberos Options](https://docs-mongodbcom-staging.corp.mongodb.com/bi-connector/anthonysansone/DOCSP-2437/reference/mongosqld.html#kerberos-options)
- [Kerberos in Security Options for Conf file](https://docs-mongodbcom-staging.corp.mongodb.com/bi-connector/anthonysansone/DOCSP-2437/reference/mongosqld.html#security-options)

---
This change is [Reviewable](https://reviewable.io/reviews/mongodb/docs-bi-connector/140)

### CODE REVIEW

File: source/includes/options-mongosqld.yaml [hunk 1]
```diff
@@ -671,4 +671,37 @@ args: <service-name>
...
+default: First IP address for :setting:`net.bindIp`.
+---
+program: mongosqld
+name: gssapiServiceName
+directive: option
+args: <service-name>
+description: |
+  .. versionadded:: 2.5
+
+  Name of the system service which runs the Kerberized :binary:`~bin.mongosqld`.
```
Comment (rychipman) [inline source/includes/options-mongosqld.yaml]: this option doesn't really have anything to do with a system service -- it is the service name component of the kerberos SPN. We can use the mongod docs for security.sasl.serviceName as a template for what this should look like

File: source/includes/options-mongosqld.yaml [hunk 2]
```diff
@@ -671,4 +671,37 @@ args: <service-name>
...
 optional: true
+---
+program: mongosqld
+name: gssapiHostname
+directive: option
+args: <hostname>
+description: |
+  .. versionadded:: 2.5
+
+  Hostname that serves Kerberos.
```
Comment (rychipman) [inline source/includes/options-mongosqld.yaml]: not necessarily the hostname that serves it (there could be many of those). this option determines the hostname part of the kerberos SPN. we can use mongod's security.sasl.hostName as a template for this option

File: source/includes/options-mongosqld.yaml [hunk 3]
```diff
@@ -671,4 +671,37 @@ args: <service-name>
...
+default: mongosql
+---
+program: mongosqld
+name: mongo-gssapiServiceName
+directive: option
+args: <service-name>
+description: |
+  .. versionadded:: 2.5
+
+  Name of the system service which runs the Kerberized ``mongod``.
```
Comment (rychipman) [inline source/includes/options-mongosqld.yaml]: this is equivalent to the mongo shell's --gssapiServiceName flag. Perhaps we can use the same documentation for this flag?

**Comment by atsansone** [comment]
@rychipman : Ready for another review. Thanks!

File: source/reference/mongosqld.txt [hunk 4]
```diff
@@ -455,6 +466,9 @@ Security Options
      enabled: <boolean>
      defaultMechanism: <string>
      defaultSource: <string>
+     gssapi.
```
[the remark this answers is not in the document]
Comment (rychipman) [reply source/reference/mongosqld.txt]: ping on this change

File: source/includes/options-mongosqld.yaml [hunk 5]
```diff
@@ -690,7 +693,11 @@ args: <service-name>
 description: |
   .. versionadded:: 2.5
 
-  Name of the system service which runs the Kerberized :binary:`~bin.mongosqld`.
+  Registered name of the service using SASL. This option allows you to
```
Comment (rychipman) [inline source/includes/options-mongosqld.yaml]: let's just remove the first sentence here (`Registered name of the service using SASL.`)

**Pull request merged**